include/xsimd/math/xsimd_rounding.hpp

```diff
@@ -157,10 +157,10 @@ namespace xsimd
         return s ^ select(v < t2n, d0 - t2n, v);
     }
 
-    template <>
-    inline batch<float, 4> trunc(const batch<float, 4>& x)
+    template <class T, std::size_t N>
+    inline batch<T, N> trunc(const batch<T, N>& x)
```

Comment:
Why this change? This function is the specialization of `trunc` for float, then the specialization for double is just below (if it exists).